# CSE3CI Lab 1: Solutions to the Exercise

**Question 1**: <br>Assume a kid Bob is 10 years old. To create this profile information for Bob , now assign a value of "Bob" to the variable *name*, and assign a value of 10 to the variable *age*, and print the variable type for both name and age

**Question 2**: <br> print the *name* and *age* with sprintf style string formatting, the print result should be "Bob is 10 years old."

#### Now let's do some exercise for list, dictionary, functions and visualization

**Question 3**: <br>create 3 dictionaries for three students in a class. The dictionary for each student have four keys: name, homework, quizzes and tests. Since there are more than one homework, quiz and test, so you will need to use list to record them.
<br><br>
The first student is Cathy. The marks for the homework are 90.0, 97.0, 75.0 and 92.0. The marks for the quizzes are 88.0, 40.0 and 94.0. The marks for the test are 75.0 and 90.0.
<br><br>
The second student is Alice. The marks for the homework are 100.0, 92.0, 98.0 and 100.0. The marks for the quizzes are 82.0, 83.0 and 91.0. The marks for the test are 89.0 and 97.0.
<br><br>
The third student is Tyler. The marks for the homework are 0.0, 87.0, 75.0 and 22.0. The marks for the quizzes are 0.0, 75.0 and 78.0. The marks for the test are 100.0 and 100.0.
<br><br>Now create three dictionaries for the three students.

**Question 4**: <br>create a gradebook for the three students using a list

**Question 5**: <br> for each student in your students list, print out that student's data, as follows:
<br><br>
print the student's name
<br>
print the student's homework
<br>
print the student's quizzes
<br>
print the student's tests

**Question 6**:
<br>Plot the four homework marks for Cathy using red line and circle marker
<br> Tips: you can create the x axis as the order of the homework

**Question 7**: <br> Write a function *average* that takes a list of numbers and returns the average.
<br><br>Define a function called average that has one argument, numbers.
<br>
Inside that function, call the built-in sum() function with the numbers list as a parameter. Store the result in a variable called total.
<br>
Use float() to convert total and store the result in total.
<br>
Divide total by the length of the numbers list. Use the built-in len() function to calculate that.
<br>
Return that result.

validate your function to see if it works as expected using the following:

In [9]:
print(average([10.0, 50.0, 90.0]))

50.0


**Question 8**: <br> Write a function called get_average that takes a student dictionary (like cathy, alice, or tyler) as input and returns his/her weighted average.
<br><br>Define a function called get_average that takes one argument called student.
<br>Make a variable homework that stores the average()(this is the function that you just created) of student["homework"].
<br>Repeat the last step for "quizzes" and "tests".
<br>Multiply the 3 averages by their weights and return the sum of those three. Homework is 10%, quizzes are 30% and tests are 60%.

validate your function to see if it works as expected using the following:

In [11]:
print(get_average(cathy))

80.55


**Question 9**:
<br>Define a new function called get_letter_grade that has one argument called score. Expect score to be a number.
<br>Inside your function, test score using a chain of if: / elif: / else: statements, like so:
<br><br>if score is 90 or above: return "A"
<br>else if score is 80 or above: return "B"
<br>else if score is 70 or above: return "C"
<br>else if score is 60 or above: return "D"
<br>Otherwise: return "F"
<br><br>Finally, validate your function. Call your get_letter_grade function with the result of get_average(lloyd). Print the resulting letter grade.

validate your function to see if it works as expected using the following:

In [13]:
print(get_letter_grade(get_average(cathy)))

B


**Question 10**:
<br>Define a function called get_class_average that has one argument, gradebook. You can expect gradebook to be a list containing your three students.
<br><br>First, make an empty list called results.
<br>For each student item in the class list, calculate get_average(student) and then call results.append() with that result.
<br>Finally, return the result of calling average() with results.

validate your function to see if it works as expected using the following:

In [18]:
print(get_class_average(gradebook))

83.86666666666666


Now, you can find that, the class average for the students is 83.87.

**Question 11**:
<br>Now use get_letter_grade and print the class's average: